test/command_callback/test_cs_mcsc_command_callbacks.vader

```diff
@@ -0,0 +1,86 @@
+Before:
+  Save g:ale_cs_mcsc_options
+  Save g:ale_cs_mcsc_source
+  Save g:ale_cs_mcsc_assembly_path
+  Save g:ale_cs_mcsc_assemblies
+
+  unlet! g:ale_cs_mcsc_options
+  unlet! g:ale_cs_mcsc_source
+  unlet! g:ale_cs_mcsc_assembly_path
+  unlet! g:ale_cs_mcsc_assemblies
+
+  runtime ale_linters/cs/mcsc.vim
+
+After:
+  Restore
+  unlet! b:ale_cs_mcsc_options
+  unlet! g:ale_cs_mcsc_source
+  unlet! g:ale_cs_mcsc_assembly_path
+  unlet! g:ale_cs_mcsc_assemblies
+  call ale#linter#Reset()
+
+Execute(Check for proper default command):
+
+  let b:command = ale_linters#cs#mcsc#GetCommand(bufnr(''))
+
+  Assert match( b:command, '^\s*cd\s\+"."\s*;\s*mcs\s\+-unsafe\s\+-out:\S\+\s\+-t:module\s\+-recurse:"\*\.cs"\s*$' ) == 0,
```

Comment:
The regular expressions here are very hard to read, so it's hard to make sense of what the tests are testing. I recommend writing tests for exact strings with `AssertEqual` instead. You will probably need to build certain parts of the strings base on the filename, etc.

Reply:
Is in this case not possible cause the GetCommand registers a temporary file with ale, which differs at every call so the assert is the only possibility to make test succeed, unless in your test machinery you have an AssertFunction or some other function which helps to handle exactly that case.

Reply:
I recommend replacing just the temporary file in the actual string with some other sequence of characters. Then you can use an exact string for the expected string.

Reply:
You could substitute `-out:[^ ]*` or something like that with `-out:TEMP` or just anything really.

Reply:
you mean 

let b:command = subsitute(b:command,'/\+tmp/\+\S\+','/tmp/some_tmp_file.tmp')

and than use this modified string in the AssertEqual

Reply:
Yeah, something like that. But bear in mind that the directory will be different from `/tmp` on Mac and Windows, so you should probably search for some characters after `'-out:'`.

Reply:
Sure i do. I finish the tests and the fixes commit and as soon as they are working properly i let you know